backend/satellite_tools/repo_plugins/yum_dnf_src.py

```diff
@@ -209,17 +209,16 @@ def setup_repo(self, repo, no_mirrors, ca_cert_file, client_cert_file, client_ke
         self.dnfbase.repos.add(repo)
         self.repoid = repo.id
         try:
-            logger = logging.getLogger('dnf')
-            logger.setLevel(logging.ERROR)
-            self.yumbase.repos[self.repoid].load()
-            logger.setLevel(logging.WARN)
-        except RepoError:
-            # Dnf bug workaround. Mirrorlist was provided but none worked. Fallback to baseurl and load again.
-            # Remove once dnf is fixed and add detection if mirrors failed.
-            logger.setLevel(logging.WARN)
-            repo.mirrorlist = ""
-            no_mirrors = True
             self.dnfbase.repos[self.repoid].load()
+            # Don't use mirrors if there are none.
+            if not self.clean_urls(self.dnfbase.repos[self.repoid]._repo.getMirrors()):
+                no_mirrors = True
+                # Reload repo just in case.
+                repo.mirrorlist = ""
+                self.dnfbase.repos[self.repoid].load()
+        except RepoError as exc:
+            raise RepoMDError(exc)
+            return
```

Comment:
Any chances to reach this `return`?

Reply:
Always? Unless I misunderstand the code. Raise only raises the error but the programme continuous (or `RepoMDError` always continues?). At least that is what it looks like on my tests. So I had to put the `return`. Otherwise there will be more errors further down because the repo didn't load properly.

Reply:
Try to run this code:
```
try:
    a = 123/0
except ZeroDivisionError:
    raise Exception("testing")
    print("test")
```

`print("test")` will never be reached

Reply:
I just re-tried without the return and the output is indeed the same. So as confused I am now, it looks like my testing was wrong....
Will remove it.
Thank you.